go/fn/context.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fn
+
+import (
+	"context"
+	"time"
+)
+
+var _ context.Context = &Context{}
+
+// TODO: Have Context implement `context.Context`.
+type Context struct {
+	ctx context.Context
```

Comment:
nit: could we embed the `context.Context` here? Any reasons not to?

Reply:
haha, because I was working on some code that literally change those functions. But you are right, the merging code should be nice and clean. 
Done!